workers/common/src/main/kotlin/common/OrtRunService.kt

```diff
@@ -489,4 +499,29 @@ class OrtRunService(
             ortRunRepository.update(ortRunId, issues = issues.asPresent())
         }
     }
+
+    /**
+     * For scan results regarding a [RepositoryProvenance], if this provenance has a VCS path, then filter
+     * out the [OrtScanResult] summaries that are not related to this path. These summaries typically contain
+     * findings about licenses, copyrights, code snippets and other information from the scan.
+     */
+    private fun filterScanResultSummary(
+        provenances: Collection<ProvenanceResolutionResult>,
+        scanResults: Set<ScanResult>
+    ): Set<OrtScanResult> {
+        val vcsPaths = provenances
+            .filterIsInstance<RepositoryProvenance>()
+            .mapNotNull { it.vcsInfo.path.takeUnless { it.isEmpty() } }
+
+        // Filter out scan result summaries outside the VCS path.
+        return scanResults.mapTo(mutableSetOf()) { scanResult ->
+            scanResult.mapToOrt().let { ortScanResult ->
+                if (scanResult.provenance is RepositoryProvenance) {
+                    ortScanResult.copy(summary = ortScanResult.summary.filterByPaths(vcsPaths))
+                } else {
+                    ortScanResult
+                }
+            }
+        }
```

Comment:
If I understand this function correctly it collects the VCS paths from all repository provenances in the scanner run in `vcsPaths` and then filters all scan results with a repository provenance by these paths.

So if repo A has VCS path "a", all scan results get filtered by path "a" no matter if they from repo A or any other repo?

Compare this to this code from ORT which does collects the VCS paths by provenance and then filter each scan result only by the paths that match the provenance:
https://github.com/oss-review-toolkit/ort/blob/254dbf96933d9514add566c62df5e41564a5612b/scanner/src/main/kotlin/Scanner.kt#L194-L212

Reply:
All in all the situation is not nice. If ORT Server sees itself as a wrapper around ORT, then ORT Server should not implement its own logic to work on the scan results, but instead use the same logic as in ORT. Okay, I can have at the code in ORT, adapt it and make a copy of the code in ORT Server that does exactly the same, but ... not nice.

Reply:
It would be better to move the code in ORT into a reusable function than to copy it.

Reply:
Did it this way now. Isolated the filter code in ORT into a Util class, and now just re-using the same in ORT Server to filter out the unwanted scan results, in the same way it is already done in ORT.